docs/examples/MondrianTreeRegressor.md

```diff
@@ -0,0 +1,273 @@
+# Intuition behind MondrianTrees
+
+This example provides an intuition behind the MondrianTreeRegressor. In particular, the key differences between an extremely randomized tree and explanations of the tree construction and prediction will be higlighted.
+
+
+```python
+# Import necessary stuff
+import numpy as np
+import matplotlib.pyplot as plt
+from sklearn.tree import ExtraTreeRegressor
+from skgarden import MondrianTreeRegressor
+from itertools import cycle
+%matplotlib inline
+```
+
+##  Extremely Randomized Tree - Recap
+
+In a tree-based algorithm, the decision rule at every node is constructed by considering a set of candidate split points and that candidate split point that minimizes a certain impurity (generally the mse in case of continuous values) is chosen. In an extremely randomized tree, the number of candidate splits `S` are limited to `n_features` and each candidate split `S[i]` is drawn unifomly from the bounds `(l_f[i], u_f[i])`.
+
+In a single extremely randomized tree, this construction might seem suboptimal but while constructing an ensemble of trees, it makes sure that each tree constructed in an independent fashion. In addition, limiting the number of candidate split points makes the tree construction very fast. Let us now generate some toy data.
```

Comment:
"it makes sure that each tree constructed in an independent fashion"

I would complete this argument by saying that decorrelating predictions in an ensemble is precisely the key factor to achieve lower generalization error.